### PR TITLE
Add test for cross namespace watch and list

### LINKED ISSUE #45511: kube-apiextensions-server needs more integration tests
There is an integration test framework for the `kube-apiextensions-server` which provides an easy-ish way to run the server standalone for testing scenarios.  See the example here: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go

This is a running list of tests that we need to have for good coverage.
 - [x] does the server come up
 - [x] does basic CRUD work (e.g. kubectl create, get, delete, label, annotate)
 - [x] does cross namespace list and watch work. @xiao-zhou
 - [x] Proper rejects for no-namespace creates. @nikhita  
 - [x] does de-registration and re-registration work @sdminonne 
 - [x] do multiple registrations work @sdminonne 
 - [x] does etcd storage function properly?  Correct serialization and key location @sdminonne 
 - [x] does discovery for kind, shortnames, etc come through correctly
 - [x] do colliding named resources work.  And when one is deleted does the gap close @nikhita
 - [x] do non-namespaced resources work correct @nikhita 
 - [x] [are self-links correct](https://github.com/kubernetes/kubernetes/issues/37622) @sdminonne 
 - [x] does finalization work.
-  [ ] do initializers work https://github.com/kubernetes/kubernetes/pull/47252 @sttts
 - [ ] e2e test that ensures that CRDs work in `kube-apiserver` https://github.com/kubernetes/kubernetes/pull/46865 @sttts 
 - [x] finalizers on CR instances should work @ash2k 

Tests to prevent regression of issues encountered previously with TPR
 - [x] does advanced CRUD work (e.g. kubectl edit, patch, apply, describe, explain, replace, watch) @nikhita 
 - [ ] does GC behave acceptably with CustomResources as owner or owned. @nikhita https://github.com/kubernetes/kubernetes/pull/46585, blocked on https://github.com/kubernetes/kubernetes/pull/46000
 - [x] [is integer data preserved](https://github.com/kubernetes/kubernetes/issues/30213) @nikhita
 - [ ] [does Helm work with CustomResources](https://github.com/kubernetes/helm/issues/1468)

@enisoc Add anything missing to the list and just ping me.
@sdminonne If you have cycles to jump in, let me know.

### PR DESCRIPTION
**Pull request by xiao-zhou**
**What this PR does / why we need it**: Add more integration test for kube-apiextensions-server

**Which issue this PR fixes** : fixes https://github.com/kubernetes/kubernetes/issues/45511

**Special notes for your reviewer**: The client with cluster scope also works, but it seems to be trivial

@deads2k

### CODE REVIEW

**Comment by deads2k** [comment]
@k8s-bot ok to test

File: staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go [hunk 1]
```diff
@@ -379,3 +379,101 @@ func TestPreserveInt(t *testing.T) {
...
+	}
+	if e, a := 0, len(initialList.(*unstructured.UnstructuredList).Items); e != a {
+		t.Errorf("expected %v, got %v", e, a)
+	}
+
+	initialListListMeta, err := meta.ListAccessor(initialList)
+	if err != nil {
+		t.Fatal(err)
+	}
+	noxuWatch, err := noxuResourceClient.Watch(metav1.ListOptions{ResourceVersion: initialListListMeta.GetResourceVersion()})
```
Comment (deads2k) [inline staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go]: can you create the namespaced watches as well and make sure that we don't end up with overlap?

**Comment by deads2k** [comment]
just the one comment, then lgtm.

File: staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go [hunk 2]
```diff
@@ -467,13 +475,43 @@ func TestCrossNamespaceListWatch(t *testing.T) {
...
+			createdObjectMeta, err := meta.Accessor(watchEvent.Object)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if e, a := ns, createdObjectMeta.GetNamespace(); e != a {
+				t.Errorf("expected %v, got %v", e, a)
+			}
+		case <-time.After(5 * time.Second):
+			if namespacedAddEvent != 1 {
+				t.Fatalf("missing watch event")
```
Comment (deads2k) [inline staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go]: The second time through the loop we expect to timeout, right?

Comment (deads2k) [inline staging/src/k8s.io/kube-apiextensions-server/test/integration/basic_test.go]: oh, gated.

**Comment by deads2k** [comment]
@k8s-bot ok to test

**Comment by deads2k** [comment]
lgtm.  Just need to squash and sort out your CLA.

Thanks for the test. :)

**Comment by xiao-zhou** [comment]
Thanks @deads2k . Since it's my first PR to k8s repo, not sure how do I add CLA?

**Comment by nikhita** [comment]
Hi @xiao-zhou! You can find the instructions to sign the CLA here: https://github.com/kubernetes/kubernetes/wiki/CLA-FAQ :)

**Comment by xiao-zhou** [comment]
Thanks @nikhita

CNCF CLA signed

**Comment by xiao-zhou** [comment]
CLA signed

**Comment by xiao-zhou** [comment]
Thanks @nikhita . CLA signed

**Comment by xiao-zhou** [comment]
Test CLA

**Pull request merged**